src/comp/comp.py

```diff
@@ -24,48 +24,50 @@ def __repr__(self):
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with 'D':
 print("Starts with D:")
-a = []
-print(a)
+a = [h.name.lower().strip("Alice Bob Charlie Eve Frank Glenn Harrison Igon").title() for h in humans]
+# a = filter(lambda h: h == 'D',humans)
 
+print(a)
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 print("Ends with e:")
-b = []
-print(b)
+b = [h.name.strip("Bob Frank Glenn Harrison Igon David") for h in humans]
+# b = filter(lambda h: min(h)== 'e',humans)
 
+print(b)
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 print("Starts between C and G, inclusive:")
-c = []
+c = ['C'> h.name[0] > 'G' for h in humans]
 print(c)
```

Comment:
You had the right idea, things were just out of order:

"c = [h.name for h in humans if h.name[0] >= 'C' and h.name[0] <= 'G']"

Make sure you put the syntax in this order:

[(what you're looking for eg. h.name), (where you're looking for it eg. for h in humans), (what conditions are needed eg. if h.name[0) >= 'C' and h.name[0] <= 'G']